test/initial_tfm.jl

```diff
@@ -0,0 +1,211 @@
+    @test isrotation(tformtest02.linear)
+
+    #with skewed spacing produces real rotations.
+    testimage5 = testimage1[:,:,-4:2:5]
+    testimage6 = testimage2[:,:,-4:2:5]
+    ps = (1, 1, 2)
+    SD =SArray{Tuple{3,3}}(Diagonal(SVector(ps) ./ minimum(ps)))
+
+    tformtest5, mm5 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int))
+    @test mm5 <1e-4
+    @test isapprox(tformtest5, mytform, atol = 1)
+    @test isrotation(tformtest5.linear)
+    @test !isrotation(SD*tformtest5.linear*inv(SD))
+
+    tformtest55, mm55 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int), initial_tfm = tformtest5)
+    @test mm55 <1e-4
+    @test isapprox(tformtest55, mytform, atol = 1)
+    @test isrotation(tformtest55.linear)
+    @test !isrotation(SD*tformtest55.linear*inv(SD))
+
+
+    #coarse and find produce real rotations
+    tformtest6, mm6 = RegisterQD.qd_rigid_coarse(testimage6, testimage5, mxshift, mxrot2, minwidth_rot2; SD=SD, print_interval=typemax(Int))
+    @test mm6 <1e-4
+    @test isrotation(tformtest6.linear)
+    @test !isrotation(SD*tformtest6.linear*inv(SD))
+
+    tformtest66, mm66 = RegisterQD.qd_rigid_coarse(testimage6, testimage5, mxshift, mxrot2, minwidth_rot2; SD=SD, print_interval=typemax(Int), initial_tfm = tformtest6)
+    @test mm66 <1e-4
+    @test isrotation(tformtest66.linear)
+    @test !isrotation(SD*tformtest66.linear*inv(SD))
+
+    tformtest7, mm7 = RegisterQD.qd_rigid_fine(testimage6, testimage5, [mxrot2...]./2, minwidth_rot2; SD=SD, print_interval=typemax(Int))
+    @test mm7 <1e-4
+    @test isrotation(tformtest7.linear)
+    @test !isrotation(SD*tformtest7.linear*inv(SD))
+
+    tformtest77, mm77 = RegisterQD.qd_rigid_fine(testimage6, testimage5, [mxrot2...]./2, minwidth_rot2; SD=SD, print_interval=typemax(Int), initial_tfm = tformtest7)
+    @test mm77 <1e-4
+    @test isrotation(tformtest77.linear)
+    @test !isrotation(SD*tformtest77.linear*inv(SD))
+    #fails due to specific error in qd_rigid_fine
+
+    # a non rigid initial_tfm does not return a rigid transformation and prints an error message (?)
+    tformtest8, mm8 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int), initial_tfm = AffineMap(SD\tformtest5.linear*SD, SD\tformtest5.translation))
+    @test mm8 <1e-4
+    @test isapprox(tformtest8, mytform, atol = 1)
+    @test !isrotation(tformtest8.linear)
+
+    mktemp() do path, io
+        redirect_stdout(io) do
+            tformtest8, mm8 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int), initial_tfm = AffineMap(SD\tformtest5.linear*SD, SD\tformtest5.translation))
+        end
+        flush(io)
+        str = read(path, String)
+        @test !isempty(str)
+    end
+
+end #TODO why do these pass when they're missing variables?
```

Comment:
They don't for me. Maybe your REPL session had those variables defined globally?